File: 3rd Sem/OOP Lab/2022_10_11/q2.cpp
#include<iostream>
using namespace std;

class Circle{
    protected:
    int rad;
    public:
    Circle(int a){
        rad = a;
    }
    void area_cir(){
        float area = 3.14 * rad* rad;
        cout<<"Area: "<<area<<endl;
    }
};

class Rectangle: public Circle{
    protected:
    int bred;
    public:
    Rectangle(int len, int bre): Circle(len){
        bred = bre;
    }
    void area_rect(){
        int area = rad * bred;
        cout<<"Area: "<<area<<endl;
    }
};

class Box: public Rectangle{
    int height;
    public:
    Box(int len, int bre, int hei): Rectangle(len, bre){
        height = hei;
    }
    void box_vol(){
        int vol = rad * bred * height;
        cout<<"Volume: "<<vol<<endl;
    }
};

int main(){
    Box b1(5,6,7);
    b1.area_cir();
    b1.area_rect();
    b1.box_vol();

    return 0;
}

File: 4th sem/WT Lab/2023_02_27/q2.java
class shape{
    double area;
    void ShowArea(){
        System.out.println("Area: "+area);
    }
}

class rectangle extends shape{
    rectangle(){
        int l = 5;
        int w = 6;
        
        super.area = l*w; 
    }
}

class circle extends shape{
    circle(){
        int r = 9;
        super.area = 3.14 * r * r;
    }
}

public class q2 {
    public static void main(String[] args){
        circle c1 = new circle();
        rectangle r1 = new rectangle();

        r1.ShowArea();
        c1.ShowArea();
    }
}
